Nanostring-Biostats/InSituType
# V2.0 (#1)

* Test with matching gene sets

* Parallelize calls to lldist by profile

* Determine number of cores for mclapply

* saving flightpath plot to /tmp/NBClust-Plots/

* better file naming with time formatting

* file name without spaces

* fixing 7X7 in for save size

* typo fix for units

* delete detritus in rescaleProfiles

* iss161 update refineClusters

* tidy vignettes

* pass check()

* make default settings more robust

and use lower max_iters in phase 1

* update docs

* partial progress

* remove geosketch redundancy

and speed up subsampling

* remove dependency on rcolorbrewer

* delete comments

* add colorCellTypes function

Function copied from ptolemy, for choosing a good initial set of colors for cell types

* document colorcelltypes

* add color info to vignettes

* add colorCellTypes function to vignettes

* fix iss165

* vignette bugfix

* copy test script for colorCellTypes from ptolemy

* Updated the temp folder creation logic

* updated package version

* Hardcode isRelease to true and comment out to deploy to all envs

* New include

* add specs and remove @exports

* add missing @params

* R CMD check fixes

* new documentation

* Remove exports

* Fix imports

* Update to MIT license

* Add to .Rbuildignore

* complete docs

* Fix documentation

* Reference global variable for R CMD check

* doc update

* Change package name to InSituType

* Change package name in Rcpp files

* Linting style changes

* Update documentation

* Unify names for vignettes

* delete settings file

* Remove writing plot to file

* Keep as one row matrix

* Render latest vignettes

* BiocCheck items

* Revert "Remove writing plot to file"

* Specify package for dataset

* Specify package for data

* Update vignettes with reference to CosMx

* Bump version

* SingleCellExperiment option for core functions

* Add dependency

* Use one core for building vignettes

* Add documentation

* keep supercluster consister when subclustering

subclustered cells are now required to fall within their original supercluster, even if another supercluster has a greater loglik

* Bump to version 0.99.3

* add link to manuscript and citation

* update README

* correct dependencies flowchart in readme

* remove dependency on lsa package in DESCRIPTION; faster way to calculate cell-level cosine similarity with celltype profiles in get_anchor_stats function

* Don't track RStudio project file

* Remove set.seed usage

* Bump version

* Change maintainer

* Add examples

* Merged PR 15109: Update azure-pipelines.yml for Azure Pipelines

Related work items: #175413

* Update azure-pipelines.yml for Azure Pipelines

* Updated trigger

* Removed 1.0 environments from main branch

* Added QA 1.1 environment

* Updated local path for copying

* Fixed variable

* Update azure-pipelines.yml for Azure Pipelines

* Fixed globExpression

* message help wip

* sync with main

* explicit convert to dgCMatrix for ensuring access to counts2@x in next line

* Re-add writing plot to disk.

* Implement OpenMP parallelization for lldist

* Don't save plot

* Added license

* Fix merge error

* Restore saving plot

* Use 64 bit integers

* filter anchor, platform adjustment

* fix umap in flightpath

* add utilities func

* Bump version to 1.1.0

* I add the updateReferenceProfiles function to estimate gene-level platform effect and rescale raw reference profile

* update function name

* replace the estimatePlatformEffects function with updated version

* auto anchor selection & reference update

* update estimatePlatformEffects function

* update the name of output list

* rename the output list name

* anchor threshold, lost genes

* anchor threshold

* fix testthat

* update mans

* outlier beta range

* lostgenes vs. blacklist

* sparse array compatibility

* net expr

* default mc.cores option

* handle glm error

* fastglm method

* remove warning_genes

* fix 0 bg in Poisson start

* fix zero bg in estimateBackground()

* Remove unneeded dependency

* Updated 13 files under /R

* Updated 7 files under /src

* Updated 32 files under /man

* added protein insituType clustering

* update InsituType.R

* updated all files

* testing folder updated

* testing folder updated

* InsituType function updated

* Deleted rescale_reference.R

* Updated gen_profiles_protein.R

* Added gen_profiles_protein_old.R

* Added 2 files to /data

* Renamed default_signature_matrix.csv to human_signature_matrix.csv

* Fix S4 method

* Remove build outputs

* Update docs

* Deleted gen_profiles_protein_old.R

* Updated nbclust.R

* insitutype function updated for protein

* gen_profiles_protein function updated

* Add lsa as dependency

* Add dependencies

* update insitutypeML

* update Rcpp functions

* update lldist function

* rcpp change

* updated all files

* Update pipeline for devnext.

* updated chooseclusternumber for protein

* Deleted NBClust-Plots

* Deleted NBClust-Plots

* added examples and commented lines

* Add back ARMA fix

* testing examples and tests files

* Fix example and vignettes

* Fix roxygen for Rcpp

* Add dependencies

* Deleted NBClust-Plots

* Re-add dependencies

* Fix errors in unit tests and examples

* Fix example

* Remove duplicates

* calling signature matrix from the package

* calling signature matrix from the package

* updated vignette

* updated the namespace

* Deleted NBClust-Plots

* Update version to 1.2.0

* Update ado pipeline.

* Merged PR 20275: Create "undefined" profile for cells with zero counts

Change behavior of InSituType in all modes--supervised, unsupervised, semi-supervised--to no longer error when there are cells with zero counts but instead set their cell type to "undefined".

For the other error in linked work item generate a more informative message indicating that the profiles do not have sufficient shared genes with panel.

Related work items: #201286

* Bump version for dev branch

* fix compatability with platform correction

* add default and limit choices for assay_type

* Bump dev version to v1.2.2

* PD updates

* update docs

* update tests

* update version and NEWS and README

* code review updates

* update news

* delete duplicated functions

* new docs

* 2nd code review updates

* update FAQ - new qc recommendation

* Merged PR 21765: handle collinearity issues with fastCohorting

handle collinearity issues with fastCohorting:
1. Reduce to 2 PC's.
2. If this fails, then try successively smaller # of cohorts with the 2 pc's.

Related work items: #211163

* Bump version for dev to 1.2.3

* bugfix

rownames(sds) was creating an error in rna mode.

* Merged PR 21963: Update pipeline for dev branch.

Update pipeline for dev branch.

Related work items: #211130

* faqs update

* updates per LW code review

* vignette finessing

* r cmd check updates

* faq updates

* remove vestigial plots

---------
